### PR TITLE
Fix Follow recenter behavior

### PR DESCRIPTION
**Pull request by afarchy**
## Overview
Fixes the re-centering logic in the Follow Solver for both Recenter() method.

## Changes
- Fixes: recenterNextUpdate wasn't being considered in several calculations. Now when this flag is set, the follow solver will instantaneously move the object in front of the user at the default distance.

Fixes: #8152

### CODE REVIEW

File: Assets/MRTK/SDK/Features/Utilities/Solvers/Follow.cs [hunk 1]
```diff
@@ -374,22 +374,29 @@ public override void SolverUpdate()
...
             ComputeOrientation(goalPosition, wasClamped, ref goalRotation);
 
-            // Avoid drift by not updating the goal position when not clamped
-            if (wasClamped)
+            if (recenterNextUpdate)
             {
-                GoalPosition = goalPosition;
+                PreviousGoalRotation = goalRotation;
+                PreviousReferencePosition = goalPosition;
+                PreviousReferenceRotation = goalRotation;
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Features/Utilities/Solvers/Follow.cs:381]: Do these not need to continue being set to `refPosition` and `refRotation`?

Comment (keveleigh) [reply Assets/MRTK/SDK/Features/Utilities/Solvers/Follow.cs:381]: Actually, look through the code...are `PreviousReferencePosition` and `PreviousReferenceRotation` actually used anywhere?

**Comment by keveleigh** [comment]
Looks like the build is failing with 

```
Assets\MRTK\SDK\Features\Utilities\Solvers\Follow.cs(382,17): error CS0618: 'Solver.SnapTo(Vector3, Quaternion)' is obsolete: 'Use SnapTo(Vector3, Quaternion, Vector3) instead.'
```

File: Assets/MRTK/SDK/Features/Utilities/Solvers/Follow.cs (hunk 1 above)
Comment (afarchy) [reply Assets/MRTK/SDK/Features/Utilities/Solvers/Follow.cs:380]: Guess not!

**Review by keveleigh — APPROVED** [review]

**Comment by dav-eloper** [comment]
@keveleigh why was Recenter() removed from OnEnable()?

**Pull request merged**